xio-core/pom.xml

```diff
@@ -243,22 +250,29 @@
 
     <dependency>
       <groupId>org.mockito</groupId>
-      <artifactId>mockito-all</artifactId>
-      <version>1.10.19</version>
+      <artifactId>mockito-core</artifactId>
+      <version>2.18.0</version>
       <scope>test</scope>
     </dependency>
 
     <dependency>
       <groupId>org.powermock</groupId>
       <artifactId>powermock-module-junit4</artifactId>
-      <version>1.6.5</version>
+      <version>2.0.0-beta.5</version>
+      <scope>test</scope>
+    </dependency>
+    
+    <dependency>
+      <groupId>org.powermock</groupId>
+      <artifactId>powermock-core</artifactId>
+      <version>2.0.0-beta.5</version>
       <scope>test</scope>
     </dependency>
 
    <dependency>
       <groupId>org.powermock</groupId>
-      <artifactId>powermock-api-mockito</artifactId>
-      <version>1.6.5</version>
+      <artifactId>powermock-api-mockito2</artifactId>
```

Comment:
i wonder if we still need powermock - https://github.com/mockito/mockito/wiki/What%27s-new-in-Mockito-2#unmockable

Reply:
I'm fine with that, we should probably have some guidelines around when/how we mock things.